src/CodeGeneratorGo.cs

```diff
@@ -64,8 +65,17 @@ public override string ImplementationFileExtension
 
             // by convention the methods in the method group with an empty
             // name go into the client template so skip them here.
+            HashSet<string> ReservedFiles = new HashSet<string>();
+            ReservedFiles.Add("models");
+            ReservedFiles.Add("client");
+            ReservedFiles.Add("version");

```

Comment:
[nit] This could be abbreviated to the following:

``` C#
HashSet<string> ReservedFiles = new HashSet<string>(){
    "models",
    "client",
    "version",
};
```

Reply:
Fixed